File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        current = 0
        maxDepth = 0
        for char in s:
            if char == '(':
                current += 1
                maxDepth = max(maxDepth, current)
            elif char == ')':
                current -= 1
        return maxDepth

File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
    let current = 0
    let maxDepth = 0
    for(let char of s){
        if (char === '('){
            current++
            maxDepth = Math.max(maxDepth, current)
        }
        else if(char === ')'){
            current--
        }
    }
    return maxDepth
};